src/main/java/org/truffleruby/language/loader/RequireNode.java

```diff
@@ -149,15 +149,17 @@ protected boolean require(VirtualFrame frame, String feature,
     private void requireCExtension(String feature, String expandedPath) {
         final FeatureLoader featureLoader = getContext().getFeatureLoader();
 
-        featureLoader.ensureCExtImplementationLoaded(feature);
+        try {
+            featureLoader.ensureCExtImplementationLoaded(feature);
 
-        if (getContext().getOptions().CEXTS_LOG_LOAD) {
-            info("loading cext module %s (requested as %s)", expandedPath, feature);
-        }
+            if (getContext().getOptions().CEXTS_LOG_LOAD) {
+                info("loading cext module %s (requested as %s)", expandedPath, feature);
+            }
 
-        try {
             featureLoader.loadCExtLibrary(expandedPath);
         } catch (Exception e) {
+            CompilerDirectives.transferToInterpreter();
```

Comment:
If the transfer is just to avoid PE issues with String.format would it be better to have a private helper method that builds up the message and has a boundary on it?

Reply:
I've used a proper boundary instead now.